### PR TITLE
Unlock js test all PublicLock versions

### PR DESCRIPTION
**Pull request by clemsos**
# Description

Modifies the test logic to make sure we actually test all Unlock versions against all PublicLock versions.

This will increase the number of tests as we will have 4 unlock contracts * 4 lock templates * 3 different locks * and about 30-40 tests for each - which means a total of 1500+ tests

NB: need a rebase when #8034 is merged

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by clemsos** [comment]
One of the main issue now is that we have to set the lock template but there is no `setLockTemplate` function in v4. I have commented out v4 for now but needs to bring it back. 

Also it is likely we can bring back v6 as there is erc1820 for hardhat

**Comment by julien51** [comment]
Let's remove support for v6 locks since there is no "alive" lock in v6 anymore.

File: packages/unlock-js/src/PublicLock/v7/purchaseKey.js [hunk 1]
```diff
@@ -1,135 +0,0 @@
-import utils from '../../utils'
```
Comment (julien51) [inline packages/unlock-js/src/PublicLock/v7/purchaseKey.js:1]: good catch!

**Review by julien51 — APPROVED** [review]
Let's ship it!

**Pull request merged**